var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -47,6 +48,7 @@ class Netcdf(Package):
 
     # Required for NetCDF-4 support
     depends_on("zlib")
+    depends_on("hdf5@:1.8.999", when='+hdf5_18')
```

Comment:
I don't think there is a need for a variant, but the change to HDF5 was introduced by request from NetCDF users quite some time ago. I want to return the ball into their court; either they keep this default, or they change it -- I can't tell because I'm not using NetCDF myself.